client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java

```diff
@@ -0,0 +1,186 @@
+
+        File confFile;
+        if (StringUtils.isNotEmpty(confLocation)) {
+            confFile = new File(confLocation, KEYCLOAK_PROPERTIES);
+
+            if (confFile.exists()) {
+                String keyConf = new String(Files.readAllBytes(confFile.toPath()), StandardCharsets.UTF_8);
+
+                JSONObject object = new JSONObject(keyConf);
+
+                REALM_ID = object.getString("realm");
+                AUTH_SERVER_URL = object.getString("auth-server-url");
+                CLIENT_ID = object.getString("resource");
+                GRANT_TYPE = "client_credentials";
+                CLIENT_SECRET = object.getJSONObject("credentials").getString("secret");
+
+                LOG.info("Keycloak conf: REALM_ID:{}, AUTH_SERVER_URL:{}, CLIENT_ID:{}, CLIENT_SECRET:{}",
+                        REALM_ID, AUTH_SERVER_URL, CLIENT_ID, CLIENT_SECRET);
+            } else {
+                throw new AtlasBaseException("Keycloak configuration file not found in location " + confLocation);
+            }
+        }
+    }
+
+    private static void init() {
+        synchronized (KeycloakClient.class) {
+            if (keycloakClient == null) {
+                keycloak = KeycloakBuilder.builder()
+                        .serverUrl(AUTH_SERVER_URL)
+                        .realm(REALM_ID)
+                        .clientId(CLIENT_ID)
+                        .clientSecret(CLIENT_SECRET)
+                        .grantType(GRANT_TYPE)
+                        .resteasyClient(new ResteasyClientBuilder().build())
+                        //.resteasyClient(ClientBuilder.newBuilder().build())
+                        .build();
+
+                keycloakClient = new KeycloakClient();
+            }
+        }
+    }
+
+    public RealmResource getRealm() {
+        return keycloak.realm(REALM_ID);
+    }
+
+    public List<UserRepresentation> getAllUsers() {
+        int start = 0;
+        int size = 100;
+
+        List<UserRepresentation> ret = new ArrayList<>();
+
+        do {
+            List<UserRepresentation> userRepresentations = getRealm().users().list(start, size);
+            ret.addAll(userRepresentations);
+            start += size;
+
+        } while (CollectionUtils.isNotEmpty(ret) && ret.size() % size == 0);
+
+        return ret;
+    }
+
+    public List<GroupRepresentation> getAllGroups() {
+
+        int start = 0;
+        int size = 100;
+
+        List<GroupRepresentation> ret = new ArrayList<>();
+
+        do {
+            List<GroupRepresentation> groupRepresentations = getRealm().groups().groups(start, size);
+            ret.addAll(groupRepresentations);
+            start += size;
+
+        } while (CollectionUtils.isNotEmpty(ret) && ret.size() % size == 0);
+
+        return ret;
+    }
+
+    public List<RoleRepresentation> getAllRoles() {
+        int start = 0;
+        int size = 100;
+
+        List<RoleRepresentation> ret = new ArrayList<>();
```

Comment:
nit: Always init `ArrayList` with size 0 => `new ArrayList<>(0);` in all places.